project-keeper/src/main/java/com/exasol/projectkeeper/JavaProjectCrawlerRunner.java

```diff
@@ -58,38 +59,40 @@ public ProjectDependencies getDependencies(final Path pomFile) {
 
     private String runCrawlerPlugin(final Path pomFile, final String goal) {
         try {
-            final Runtime rt = Runtime.getRuntime();
-            final List<String> commandParts = new ArrayList<>(List.of("mvn", "--batch-mode",
+            final List<String> commandParts = new ArrayList<>(List.of(getMavenExecutable(), "--batch-mode",
                     "-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn",
                     "com.exasol:project-keeper-java-project-crawler:" + this.ownVersion + ":" + goal, "--file",
                     pomFile.toString()));
             if (this.mvnRepositoryOverride != null) {
                 commandParts.add("-Dmaven.repo.local=" + this.mvnRepositoryOverride);
             }
 
-            final Process proc = rt.exec(commandParts.toArray(String[]::new));
+            LOGGER.fine(() -> "Executing command " + commandParts);
+            final Process proc = new ProcessBuilder(commandParts).redirectErrorStream(true).start();
+
+            CollectingConsumer streamConsumer = new AsyncStreamReader().startCollectingConsumer(proc.getInputStream());
+
             if (!proc.waitFor(90, TimeUnit.SECONDS)) {
-                final String output = readFromStream(proc.getInputStream());
-                LOGGER.log(Level.SEVERE, output);
-                final String errors = readFromStream(proc.getErrorStream());
-                LOGGER.log(Level.SEVERE, errors);
                 throw new IllegalStateException(ExaError.messageBuilder("E-PK-CORE-81")
-                        .message("Timeout while executing project-keeper-java-project-crawler.").toString());
+                        .message("Timeout while executing command {{executed command|uq}}. Output was {{output}}",
+                                commandParts, streamConsumer.toString())
+                        .toString());
             }
             final int exitCode = proc.exitValue();
-            final String output = readFromStream(proc.getInputStream());
+            final String output = streamConsumer.toString();
```

Comment:
I think the consumer sould wait until the stream is closed. Here we should wait for the reader to signal that via a mutex. Otherwise, it could happen that the reader was not scheduled recently and did not read everything.

Reply:
Good point. I refactored the code to wait until the stream is closed.